python/marvin/tools/rss.py

```diff
@@ -1,102 +1,116 @@
+    they are accessed.
 
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, input=None, filename=None, mangaid=None, plateifu=None,
+                 mode=None, data=None, release=None, autoload=True,
+                 drpall=None, download=None, nsa_source='auto'):
 
-        valid_kwargs = ['filename', 'mangaid', 'plateifu', 'mode',
-                        'drpall', 'release', 'nsa_source']
+        MarvinToolsClass.__init__(self, input=input, filename=filename, mangaid=mangaid,
+                                  plateifu=plateifu, mode=mode, data=data, release=release,
+                                  drpall=drpall, download=download)
 
-        assert len(args) == 0, 'RSS does not accept arguments, only keywords.'
-        for kw in kwargs:
-            assert kw in valid_kwargs, 'keyword {0} is not valid'.format(kw)
+        NSAMixIn.__init__(self, nsa_source=nsa_source)
 
-        self.data = None
+        #: An `astropy.table.Table` with the observing information associated
+        #: with this RSS object.
+        self.obsinfo = None
 
-        MarvinToolsClass.__init__(self, *args, **kwargs)
+        #: If True, unloaded `.RSSFiber` instances are automatically loaded
+        #: when accessed. Otherwise, they need to be loaded via `.RSSFiber.load`.
+        self.autoload = autoload
 
         if self.data_origin == 'file':
-            self._load_rss_from_file()
+            self._load_rss_from_file(data=self.data)
         elif self.data_origin == 'db':
-            self._load_rss_from_db()
+            self._load_rss_from_db(data=self.data)
         elif self.data_origin == 'api':
             self._load_rss_from_api()
 
-        _fibers = self._init_fibers()
-        list.__init__(self, _fibers)
+        Cube._init_attributes(self)
+
+        # Checks that the drpver set in MarvinToolsClass matches the header
+        header_drpver = self.header['VERSDRP3'].strip()
+        header_drpver = 'v1_5_1' if header_drpver == 'v1_5_0' else header_drpver
+        assert header_drpver == self._drpver, ('mismatch between cube._drpver={0} '
+                                               'and header drpver={1}'.format(self._drpver,
+                                                                              header_drpver))
 
-        # TODO: check that the drpver of the loaded data matches the one in the object.
+        # Inits self as an empty list.
+        list.__init__(self, [])
+        self._populate_fibres()
+
+    def _set_datamodel(self):
+        """Sets the datamodel for DRP."""
+
+        self.datamodel = datamodel_rss[self.release.upper()]
+        self._bitmasks = datamodel_rss[self.release.upper()].bitmasks
 
     def __repr__(self):
         """Representation for RSS."""
 
         return ('<Marvin RSS (mangaid={self.mangaid!r}, plateifu={self.plateifu!r}, '
                 'mode={self.mode!r}, data_origin={self.data_origin!r})>'.format(self=self))
 
+    def __getitem__(self, fiberid):
```

Comment:
what I did with `Plate`, as that was a list of `Cubes`, was override `__getitem__` to get an item by generic list index, or plateifu, and made it a FuzzyList.  You might want to do the same here with RSS to allow selection by exposure number and/or MJD, in addition to fiberid.  To be a bit more flexible.

Reply:
In principle I like the idea of allowing to select fibres, or groups of them, by other parameters, but I think not in the `__getitem__`. I feel it could become a bit complicated because it's all numbers. I guess you could differentiate between an MJD and a fiberid but I think that's best implemented as additional methods (e.g., `get_mjds()`).

Reply:
I guess as it's a list of fibers, slicing by fiber id.  I think selecting by exposure number or MJD will always return sets of fibers, so having those as methods would be a good idea.  I'd say include a way of getting lists of available MJDs and exposure numbers from the RSS, as well as selecting fiber sets given an input MJD or exposure number.

Reply:
I think we can add a single `select_fibers` or similar, that accepts an number of arguments (`mjd=`, `exposure_no=`, `set=`, etc) and returns the `RSSFibers` that match that selection. That's not difficult to implement.

Reply:
Done.